modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/builder/OAuthAuthorizationCodeDancerBuilder.java

```diff
@@ -29,8 +29,11 @@
    * @param encodeClientCredentialsInBody If @{code true}, the client id and client secret will be sent in the request body.
    *        Otherwise, they will be sent as basic authentication.
    *
+   * @deprecated since 4.2.0. Use {@link OAuthAuthorizationCodeDancerBuilder#withClientCredentialsIn(ClientCredentialsLocation)} instead.
+   *
    * @return this builder
    */
+  @Deprecated
   default OAuthAuthorizationCodeDancerBuilder encodeClientCredentialsInBody(boolean encodeClientCredentialsInBody) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Do not forget to remove this deprecated code someday. ([squid:S1133](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1133))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7676&issues=AWjnfbgqotIXnmsUpaHQ&open=AWjnfbgqotIXnmsUpaHQ)

[ISSUE_KEY]: AWjnfbgqotIXnmsUpaHQ